Design configuration class for integration of HashiCorp Vault With Spring Boot Application

gitCarrot opened the pull request:
# Ticket Link

[Ticket-1833](https://uhawaii.atlassian.net/browse/GROUPINGS-1833)

# List of squashed commits

- Implement VaultConfig using spring cloud vault
- Fix bean creation error of VaultConfig and completly separate existed local dev environment and docker for running on the both, and modify the tests for vault config class
- Fix code style
- Update the version of dependencies for cloud vault config after checking compatibility with current spring boot version
- Make extra configs to manage the environments that doesn't have active vault on the localhost or dockerhost
- Remove vault core dependency from pom.xml
-  Move token property from source code to docker compose to avoid the situation that
 develoepr push the code with token by mistake, and disable vault in the localhost for the people who don't want to auto create vault bean without installing vault in their os
- Fix property of injection value in the vault config from the secret engine of the vaule



# Test Checklist

- [x] Exhibits Clear Code Structure:
- [x] Project Unit Tests Passed:
- [x] Project Integration Tests Passed:
- [x] Executes Expected Functionality:
- [x] Tested For Incorrect/Unexpected Inputs:

mhodgesatuh reviewed (commented):
Would like to understand how we might want to use the properties file to configure the URL to the vault.  And how the token for vault access is being provided.

gitCarrot commented:
@mhodgesatuh I just relocated the Vault token property from the application properties file of the API project to the Docker Compose environment variables. This change ensures that developers do not accidentally commit their tokens to the main branch.

The dependency I incorporated for retrieving Vault secrets is provided by Vault itself. It offers a simpler interface for accessing secrets compared to the original core Vault dependency, which involves numerous complex classes. By simply adding this dependency and configuring a few properties, developers can easily fetch secrets from Vault and directly set them in the Grouper client configuration properties.

One significant advantage of this dependency is that it eliminates the need for maintaining a separate bean for Vault operations. Additionally, I have moved all Vault-related properties into the Docker Compose setup (but in this ticket, I still keep some of the vault properties in the source code ) . I have verified that all these properties function correctly within Docker Compose, enhancing the security of the project.

gitCarrot commented:
There are ootb controller and controller test that should be contained in the last ticket. 
And I'm trying to figure out rebase error for squashing commits

JorWo commented on src/main/java/edu/hawaii/its/api/controller/OotbRestController.java:
Huh I thought this was in the API already.

gitCarrot replied:
I know.. haha

JorWo commented on src/main/resources/application-dockerhost.properties:
```diff
@@ -2,3 +2,12 @@
+spring.cloud.vault.uri=https://localhost:8200
+spring.cloud.vault.connection-timeout=5000
+spring.cloud.vault.read-timeout=15000
+
+
```
Make sure to delete these extra spaces.

mhodgesatuh reviewed (approved):
It is not clear to me why OOTB is also included in this PR.

gitCarrot commented:
> It is not clear to me why OOTB is also included in this PR.

@mhodgesatuh For some reason, the updates of OOTB that were supposed to be included in the last PR were not applied. Jordan also didn't realize it was not pushed if you see his comment above.
